src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderDistributionUIConfig.java

```diff
@@ -54,9 +53,15 @@ public ProviderDistributionUIConfig(final String label, final String urlName, fi
             Collectors.toList()), this::validateNotificationTypes);
         final ConfigField formatField = SelectConfigField.createRequired(KEY_FORMAT_TYPE, "Format", provider.getSupportedFormatTypes().stream().map(FormatType::name).collect(Collectors.toList()));
 
-        return Arrays.asList(notificationTypesField, formatField);
+        final List<ConfigField> configFields = List.of(notificationTypesField, formatField);
+        final List<ConfigField> providerDistributionFields = createProviderDistributionFields();
+        providerDistributionFields.addAll(configFields);
```

Comment:
We have to be careful using `.addAll(...)` on lists we don't create. It could very easily be implemented as an unmodifiable list (e.g. `List.of(...)`).
An alternative could be:
```java
Streams.concat(providerDistributionFields, configFields).collect(Collectors.toList())
```

Reply:
We should not be modifying the list that is provided as a parameter if we are returning the modified list. What we could do is return a new list and add the things we want to the new list